src/adsbinarynode.jl

```diff
@@ -0,0 +1,402 @@
+# This file is part of the TaylorIntegration.jl package; MIT licensed
+
+using StaticArrays: SVector
+import AbstractTrees: HasNodeType, NodeType, ParentLinks, StoredParents, TreeIterator,
+                      children, nodetype, nodevalue, parent, printnode, PreOrderDFS
+import Base: IteratorEltype, HasEltype, in, eltype, split
+
+if !isdefined(Base, :isnothing)        # Julia 1.0 support
+    using AbstractTrees: isnothing
+end
+
+export ADSDomain
+
+"""
+    ADSDomain{N, T <: Real}
+
+`N` dimensional box used in Automatic Domain Splitting to represent the
+domain of `N` jet transport variables.
+
+# Fields
+
+- `lo::SVector{N, T}`: lower bounds.
+- `hi::SVector{N, T}`: upper bounds.
+"""
+struct ADSDomain{N, T <: Real}
+    lo::SVector{N, T}
+    hi::SVector{N, T}
+    # Inner constructor
+    function ADSDomain{N, T}(lo::SVector{N, T}, hi::SVector{N, T}) where {N, T <: Real}
+        @assert all(hi .> lo) "Each upper bound must be greater than the respective lower bound"
+        new{N, T}(lo, hi)
+    end
+end
+
+# Outer constructors
+ADSDomain(lo::SVector{N, T}, hi::SVector{N, T}) where {N, T <: Real} = ADSDomain{N, T}(lo, hi)
```

Comment:
I'd suggest to add a couple of unit tests for the constructors in lines 36 and 38

Reply:
Done.